packages/core/src/domain/session/sessionState.ts

```diff
@@ -1,22 +1,40 @@
-import { isEmptyObject } from '../../tools/utils/objectUtils'
 import { objectEntries } from '../../tools/utils/polyfills'
 import { dateNow } from '../../tools/utils/timeUtils'
 import { SESSION_EXPIRATION_DELAY } from './sessionConstants'
 
 const SESSION_ENTRY_REGEXP = /^([a-z]+)=([a-z0-9-]+)$/
 const SESSION_ENTRY_SEPARATOR = '&'
 
+export const enum SessionExpiredReason {
+  UNKNOWN = '0',
+}
+
 export interface SessionState {
   id?: string
   created?: string
   expire?: string
   lock?: string
+  expired?: SessionExpiredReason
 
   [key: string]: string | undefined
 }
 
+export function getInitialSessionState(): SessionState {
+  return {
+    expired: SessionExpiredReason.UNKNOWN,
+  }
+}
+
+export function isSessionInitialized(session: SessionState) {
+  return session.id !== undefined || session.expired !== undefined
+}
+
 export function isSessionInExpiredState(session: SessionState) {
-  return isEmptyObject(session)
+  // // an expired session is `{expired = '0'}` or `{expired = '0', lock = whatever}`
+  return (
+    (Object.keys(session).length === 1 && session.expired !== undefined) ||
+    (Object.keys(session).length === 2 && session.expired !== undefined && session.lock !== undefined)
+  )
 }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏ ‏I am a bit confused with the relationship between expired / initial / initialized 😕
Could we clarify that somehow?

Reply:
Yeah, I agree this is confusing. I'm not sure how to improve that though:

- initial state: `{expired: 0}`
- initialized: I wish I could call it `!isEmpty(sessionState)`, however, this is not true. There is some cases where the cookie is only `rum=0`
- expired state: here again I wish I could only check if `expired` is defined. But as I [commented here](https://github.com/DataDog/browser-sdk/pull/2673#discussion_r1555974777) this is not true.

The session management is definitively confusing, and would be worth refactoring to make changes easier and safer (Although I'm not totally sure how). 

Do you have any suggestions that would fit in the scope of this PR?